typify-impl/src/lib.rs

```diff
@@ -221,6 +227,94 @@ impl TypeSpace {
+                                info!("boxing option");
+                                let box_id = box_id.get_or_insert_with(|| self.id_to_box(&type_id));
+                                *option_type_id = box_id.clone();
+                            }
+                        }
+                        // TODO this is unconditional
+                        let _ = self
+                            .id_to_entry
+                            .insert(prop.type_id.clone(), prop_type_entry);
+                    }
+                }
+            } else if let TypeEntryDetails::Enum(type_entry_enum) = &mut type_entry.details {
+                for variant in &mut type_entry_enum.variants {
+                    match &mut variant.details {
+                        VariantDetails::Simple => {}
+                        VariantDetails::Tuple(vec_type_id) => {
+                            for tuple_type_id in vec_type_id {
+                                if *tuple_type_id == type_id {
+                                    info!("boxing tuple type");
+                                    let box_id =
+                                        box_id.get_or_insert_with(|| self.id_to_box(&type_id));
+                                    *tuple_type_id = box_id.clone();
+                                }
+                            }
+                        }
+                        VariantDetails::Struct(vec_struct_property) => {
+                            for struct_property in vec_struct_property {
+                                let vec_type_id = &mut struct_property.type_id;
+                                if *vec_type_id == type_id {
+                                    info!("boxing variant struct type");
+                                    let box_id =
+                                        box_id.get_or_insert_with(|| self.id_to_box(&type_id));
+                                    *vec_type_id = box_id.clone();
+                                } else {
+                                    let mut prop_type_entry =
+                                        self.id_to_entry.get_mut(vec_type_id).unwrap().clone();
+
+                                    if let TypeEntryDetails::Option(option_type_id) =
+                                        &mut prop_type_entry.details
+                                    {
+                                        if *option_type_id == type_id {
+                                            info!("boxing option in struct property");
+                                            let box_id = box_id
+                                                .get_or_insert_with(|| self.id_to_box(&type_id));
+                                            *option_type_id = box_id.clone();
+                                        }
+                                    }
+
+                                    // TODO unconditional
+                                    let _ = self
```

Comment:
I don't think we need to explicitly ignore the return, but it certainly doesn't hurt.

Reply:
This has moved to the end of add_ref_types, and I removed the comment. The intention is clear from the code that we're modifying existing entries in id_to_entry unconditionally.